### PR TITLE
[WIP] Use proxy when building Velum image

### PR DESCRIPTION
**Pull request by kiall**
When building with a proxy, pass the proxy info into the Velum container
for when it's installing gems.

### CODE REVIEW

**Comment by kiall** [comment]
Verified via proxy logs this was used.

> caasp-proxy-01:~ # grep rubygems /var/log/squid/access.log | perl -pe 's/\d+/localtime($&)/e'
> Tue Feb 27 19:37:31 2018.634    644 10.84.73.197 TCP_TUNNEL/200 160953 CONNECT rubygems.org:443 - HIER_DIRECT/151.101.194.2 -
> Tue Feb 27 19:37:31 2018.636    829 10.84.73.197 TCP_TUNNEL/200 86567 CONNECT rubygems.org:443 - HIER_DIRECT/151.101.194.2 -
> Tue Feb 27 19:37:31 2018.640   1027 10.84.73.197 TCP_TUNNEL/200 69130 CONNECT rubygems.org:443 - HIER_DIRECT/151.101.194.2 -
> Tue Feb 27 19:37:31 2018.641   1462 10.84.73.197 TCP_TUNNEL/200 92228 CONNECT rubygems.org:443 - HIER_DIRECT/151.101.194.2 -
> Tue Feb 27 19:37:35 2018.851   4618 10.84.73.197 TCP_TUNNEL/200 9295330 CONNECT rubygems.org:443 - HIER_DIRECT/151.101.194.2 -
> Tue Feb 27 19:37:44 2018.352   1055 10.84.73.197 TCP_TUNNEL/200 236363 CONNECT rubygems.org:443 - HIER_DIRECT/151.101.194.2 -
> Tue Feb 27 19:37:44 2018.356    760 10.84.73.197 TCP_TUNNEL/200 37841 CONNECT rubygems.org:443 - HIER_DIRECT/151.101.194.2 -

File: caasp-kvm/tools/build-velum-image [hunk 1]
```diff
@@ -4,7 +4,8 @@ set -euo pipefail
 DIR="$( cd "$( dirname "$0" )" && pwd )"
 
 VELUM_DIR="$1"
-IMAGE=${2:-}
+IMAGE="$2"
+PROXY=${3:-}
```
Comment (stefsuse) [inline caasp-kvm/tools/build-velum-image]: quotes?

File: caasp-kvm/tools/build-velum-image [hunk 2]
```diff
@@ -36,7 +37,12 @@ build_fresh_image() {
   # Reuse the gems already vendored into the devel image if possible.
   # This will pull only the development gems and the ones changed on master.
   # This greatly reduces the creation time of the image
-  local CONTAINER_BUILD=`docker run -v $VELUM_DIR:/srv/velum-latest -d sles12/velum-development:0.0 \
+  local PROXY_FLAGS=""
+  if [ -n "$PROXY" ]; then
+    PROXY_FLAGS="-e http_proxy=$PROXY -e https_proxy=$PROXY -e HTTP_PROXY=$PROXY -e HTTPS_PROXY=$PROXY"
+  fi
+
+  local CONTAINER_BUILD=`docker run -v $VELUM_DIR:/srv/velum-latest -d $PROXY_FLAGS sles12/velum-development:0.0 \
```
Comment (stefsuse) [inline caasp-kvm/tools/build-velum-image]: just a preference, but $() is more readable and flexible than backticks

**Comment by MalloZup** [comment]
@stefsuse  what we do with this PR? should we close it

**Comment by dannysauer** [comment]
/kicked off the CI stuff again to ensure current compliance

**Comment by MalloZup** [comment]
@dannysauer ok i see thx for context. feel free to merge once CI pass

**Comment by dannysauer** [comment]
Well, CI failed.  So let's try pulling in current CI fixes by rebasing on upstream/master, and see if that behaves any better... :)

**Comment by dannysauer** [comment]
Switch to work-in-progress while I try sending incrementally-changed versions through CI to figure out what's causing the failure. :/

**Comment by MalloZup** [comment]
@dannysauer feel free to merge this and rebase/squash commits if needed

**Comment by dannysauer** [comment]
This needs rewritten, and I'm a little concerned about how setting these environment variables up may affect communication between Velum and Dex if they propagate down.

**Pull request closed without merging**